clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -272,6 +272,7 @@ private boolean trySend(long now) {
 
     private void clientPoll(long timeout, long now) {
         client.poll(timeout, now);
+        now = time.milliseconds();
```

Comment:
@hachikuji Yeah, the class definitely needs a new name given that `Time` is already taken. `Now` makes sense.

wrt where the burden lies, this is just the basic tradeoff between passing an object that can be updated (i.e. `Now`) vs the value directly (i.e. `long now`). The former requires that a few methods make sure they invoke `updateNow` -- usually ones that drive the main event loop or call anything that directly blocks for some time. The latter requires a lot more methods to make sure that if they directly or indirectly invoke something that can block, they need to make sure they update their `now` value. I agree neither is perfect, but the former seems a lot less error prone. I think if we did a survey of the methods that are using the current approach and indirectly invoke a blocking call and then reuse the `now` value (like `pollUnsentRequests` mentioned above), we'll find that there's actually quite a bit of code handling this incorrectly now.

Of course the other important question is how many of those erroneous uses of the previous `now` value have any real impact...